core/src/main/java/io/grpc/inprocess/InProcessServer.java

```diff
@@ -40,11 +40,16 @@
   private static final ConcurrentMap<String, InProcessServer> registry
       = new ConcurrentHashMap<>();
 
-  static InProcessServer findServer(String name) {
```

Comment:
The possibility of creating a base AbstractInProcessSocketAddress class with package visible findServer(), registerServer() and clearServer() methods occurred to me. 

That would allow this method, registerServer() and unregisterServer() below to just be methods on the address, removing the need for instance checks.

Reply:
Optimizing this method seems like a micro-optimization. If this method is the only one that benefits, then avoiding public API seems clearly superior. If we think _user's_ code would improve, then common base class is more interesting. Seems we can always do it in the future, so let's favor less API for now.

Your suggestion would also mean moving the registry to InProcessSocketAddress, which propagates the "strangeness" of anonymous socket address. Seems counter to our preferences.